### ISSUE TITLE
Is there any convenient way to do spectrum normalization?

### ISSUE THREAD

**Issue by mingyr** [deepmind/sonnet#99]
Dear Deepminder:

For generative adversarial network, there introduces a normalization called spectrum normalization (ref: https://arxiv.org/abs/1802.05957). 

However I find according to the knowledge of me, it is a little hard for me to incorporate it into Sonnet.

For example, the following is a code snippet:
```

def discriminator(images, reuse_variables=None):
    with tf.variable_scope(tf.get_variable_scope(), reuse=reuse_variables) as scope:
        # First convolutional and pool layers
        # This finds 32 different 5 x 5 pixel features
        d_w1 = tf.get_variable('d_w1', [5, 5, 1, 32], initializer=tf.truncated_normal_initializer(stddev=0.02))
        d_b1 = tf.get_variable('d_b1', [32], initializer=tf.constant_initializer(0))
        d1 = tf.nn.conv2d(input=images, filter=spectral_norm(d_w1), strides=[1, 1, 1, 1], padding='SAME')
        d1 = d1 + d_b1
        d1 = tf.nn.relu(d1)
        d1 = tf.nn.avg_pool(d1, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')
 
```

Please pay special attention to the usage below:
`d1 = tf.nn.conv2d(input=images, filter=spectral_norm(d_w1), strides=[1, 1, 1, 1], padding='SAME')
`

It means a altered version of weight is used. But I checked snt.Conv2D, seems the weight is intractable due to internal management.

So could I be given some suggestion regarding such a situation? Thanks a lot.
